### PR TITLE
THEMES-739 top table list v2

### PR DESCRIPTION
**Pull request by matthewroach**
## Description

Convert Top Table List to v2 architecture

## Jira Ticket

- [THEMES-739](https://arcpublishing.atlassian.net/browse/THEMES-739)

## Test Steps

Feature Pack PR -

Use Themes Internal Sandbox for content base

1. Checkout this branch `git checkout THEMES-739-top-table-list`
2. Checkout the feature pack branch `THEMES-739-top-table-list`
3. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/top-table-list-block`
4. Set up a new page with the top table list block
5. Configure block with content 
    * Display content info = `ans-feed`
    * content source = `story-feed-query`
    * offset = `1`
    * query = `type:story`
    * size = `20`
6. Configure the block to display a different number of story types
7. Validate output is as before
 
## Dependencies or Side Effects

Feature Pack PR - https://github.com/WPMedia/arc-themes-feature-pack/pull/364
Manifest PR - https://github.com/WPMedia/arc-themes-manifests/pull/149

## Author Checklist

- [x] Confirmed all the test steps a reviewer will follow above are working.
- [x] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [x] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [x] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [x] Confirmed this PR has unit test files
  - [x] Ran `npm run test`, made sure all tests are passing
  - [x] If the amount of work to write unit tests for this change are excessive,
        please explain why (so that we can fix it whenever it gets refactored).
- [x] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

### CODE REVIEW

File: blocks/top-table-list-block/features/top-table-list/_children/extra-large.test.jsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import mockData from "../../../mock-data";
+
+import ExtraLarge from "./extra-large";
+
+jest.mock("@wpmedia/engine-theme-sdk", () => ({
+	...jest.requireActual("@wpmedia/engine-theme-sdk"),
+}));
```
Comment (nschubach) [inline blocks/top-table-list-block/features/top-table-list/_children/extra-large.test.jsx]: This seems useless?

File: blocks/top-table-list-block/features/top-table-list/_children/large.jsx [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+	const embedMarkup = playVideoInPlaceLG && getVideoFromANS(element);
+	const promoImageURL = showImageLG && (getImageFromANS(element)?.url || fallbackImage);
+
+	const imageParams = element
+		? {
+				ansImage: getImageFromANS(element),
+				alt: element?.headlines?.basic || "",
+				aspectRatio: imageRatioLG,
+				resizedOptions: {
+					smart: true,
```
Comment (nschubach) [inline blocks/top-table-list-block/features/top-table-list/_children/large.jsx]: Don't we need an `auth` token here?

Comment (nschubach) [reply blocks/top-table-list-block/features/top-table-list/_children/large.jsx]: for all the images, that is...

**Comment by nschubach** [comment]
I'm not seeing much to bring up here except the testing import and auth token question.

File: blocks/top-table-list-block/features/top-table-list/_children/large.jsx (hunk 2 above)
Comment (matthewroach) [reply blocks/top-table-list-block/features/top-table-list/_children/large.jsx]: It's handled within the `<Image />` component now if you use the `ansImage` prop

File: blocks/top-table-list-block/features/top-table-list/_children/extra-large.test.jsx (hunk 1 above)
Comment (matthewroach) [reply blocks/top-table-list-block/features/top-table-list/_children/extra-large.test.jsx]: Pushed a fix

**Comment by malavikakoppula** [comment]
Just curious, Large-promo is not displaying lead-art image due to no src assigned to it. Where is it supposed to be handled?

**Comment by matthewroach** [comment]
> Just curious, Large-promo is not displaying lead-art image due to no src assigned to it. Where is it supposed to be handled?

@malavikakoppula Can you provide more information, or link to chromatic on what the issue is you are seeing?

**Comment by malavikakoppula** [comment]
> > Just curious, Large-promo is not displaying lead-art image due to no src assigned to it. Where is it supposed to be handled?
> 
> @malavikakoppula Can you provide more information, or link to chromatic on what the issue is you are seeing?

**Review by nschubach — DISMISSED** [review]

**Review by nschubach — DISMISSED** [review]

**Comment by matthewroach** [comment]
@haganka - I've updated the code based on comments in chromatic, you'll also see updates to Large Promo as noticed they were affected, and the current snaps shots are incorrect anyhow

**Comment by matthewroach** [comment]
@haganka - All updated now - and chromatic comments addressed

**Pull request merged**